staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -789,6 +789,18 @@ func ValidateCustomResourceDefinitionOpenAPISchema(schema *apiextensions.JSONSch
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("x-kubernetes-preserve-unknown-fields"), *schema.XPreserveUnknownFields, "must be true or undefined"))
 	}
 
+	if schema.XMapType != nil && schema.Type != "object" {
+		if len(schema.Type) == 0 {
+			allErrs = append(allErrs, field.Required(fldPath.Child("type"), "must be object if x-kubernetes-map-type is set"))
```

Comment:
hm, tiny optional nit, personally I'd flip this error report-- x-kubernetes-map-type may only be set on objects (type == "object").

Reply:
Happy to do that, this format is being consistent with the `if schema.XListType != nil && schema.Type != "array" {` error below though. So if we want to, may be better change all in one go to a preferred format in a different PR